### DISCUSSION TITLE
does dexidp provide FIPS-compatible builds?

### DISCUSSION THREAD

**Discussion by qudongfang** [dexidp/dex#3271] [Q&A] [answered]
does dexidp provide FIPS-compatible builds?

> The cryptographic routines in the Go standard library cannot be FIPS certified and must instead rely on an external cryptographic implementation which can be FIPS certified.

https://groups.google.com/g/golang-dev/c/gtyWcJabdlg

**Comment by nabokihms** [comment, marked as answer]
We discussed it [on Slack](https://cloud-native.slack.com/archives/C01HF2G1R34/p1705065388203909).

Basically, to be FIPS-compatible, the only requirement is to build Dex with the particular version of the compiler, e.g., [microsoft/go](https://github.com/microsoft/go).

Dex is not aimed to use any other builds of go than the main build provided by the go team. Yet, if you want to create and build a fork yourself, we can offer you our support.
